src/coreclr/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -90,9 +90,18 @@ COOP_PINVOKE_HELPER(int32_t, RhGetModuleFileName, (HANDLE moduleHandle, _Out_ co
 
 COOP_PINVOKE_HELPER(void, RhpCopyContextFromExInfo, (void * pOSContext, int32_t cbOSContext, PAL_LIMITED_CONTEXT * pPalContext))
 {
-    UNREFERENCED_PARAMETER(cbOSContext);
     ASSERT((size_t)cbOSContext >= sizeof(CONTEXT));
+    memset(pOSContext, 0, cbOSContext);
     CONTEXT* pContext = (CONTEXT *)pOSContext;
+
+#ifndef HOST_WASM
+    if (TryPopulateControlSegmentRegisters(pContext))
+    {
+        pContext->ContextFlags |= CONTEXT_CONTROL;
```

Comment:
CONTEXT_CONTROL also represents Rsp and Rip so it should be always set.

Reply:
I changed it to always set CONTEXT_CONTROL. Is this what you had in mind?